File: Android/scenes/ktv/src/main/java/io/agora/scene/ktv/widget/lrcView/MyScoringView.java
package io.agora.scene.ktv.widget.lrcView;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.LinearGradient;
import android.graphics.Shader;
import android.util.AttributeSet;

import androidx.annotation.Nullable;

import io.agora.karaoke_view.v11.ScoringView;
import io.agora.scene.ktv.R;

/**
 * 音高展示
 */
public class MyScoringView extends ScoringView {
    private LinearGradient mStartLineLinearGradient;

    /**
     * Instantiates a new My scoring view.
     *
     * @param context the context
     */
    public MyScoringView(Context context) {
        super(context);
    }

    /**
     * Instantiates a new My scoring view.
     *
     * @param context the context
     * @param attrs   the attrs
     */
    public MyScoringView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    /**
     * Instantiates a new My scoring view.
     *
     * @param context      the context
     * @param attrs        the attrs
     * @param defStyleAttr the def style attr
     */
    public MyScoringView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    /**
     * Instantiates a new My scoring view.
     *
     * @param context      the context
     * @param attrs        the attrs
     * @param defStyleAttr the def style attr
     * @param defStyleRes  the def style res
     */
    public MyScoringView(Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
        super.onLayout(changed, left, top, right, bottom);
        if (changed) {
            int colorWhite35 = getResources().getColor(R.color.white_35_percent);
            int colorWhite = getResources().getColor(R.color.white);
            mStartLineLinearGradient = new LinearGradient(0, 0, 0, bottom - top, new int[]{colorWhite35, colorWhite, colorWhite35}, null, Shader.TileMode.CLAMP);
        }
    }

    @Override
    protected void drawOverpastWallAndStartLine(Canvas canvas) {
        drawOverpastWall(canvas);

        // Same as drawStartLine, but with extra gradient color on it
        mStartLinePaint.setShader(mStartLineLinearGradient);
        mStartLinePaint.setColor(mLocalPitchIndicatorColor);
        mStartLinePaint.setAntiAlias(true);
        mStartLinePaint.setStrokeWidth(3);
        canvas.drawLine(mCenterXOfStartPoint, 0, mCenterXOfStartPoint, getHeight(), mStartLinePaint);
    }
}

File: Android/scenes/cantata/src/main/java/io/agora/scene/cantata/ui/widget/lrcView/MyScoringView.kt
package io.agora.scene.cantata.ui.widget.lrcView

import android.content.Context
import android.graphics.Canvas
import android.graphics.LinearGradient
import android.graphics.Shader
import android.util.AttributeSet
import io.agora.karaoke_view.v11.ScoringView
import io.agora.scene.cantata.R

class MyScoringView : ScoringView {
    private var mStartLineLinearGradient: LinearGradient? = null

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(
        context,
        attrs,
        defStyleAttr,
        defStyleRes
    )

    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
        super.onLayout(changed, left, top, right, bottom)
        if (changed) {
            val colorWhite35 = resources.getColor(R.color.white_35_percent)
            val colorWhite = resources.getColor(R.color.white)
            mStartLineLinearGradient = LinearGradient(
                0f,
                0f,
                0f,
                (bottom - top).toFloat(),
                intArrayOf(colorWhite35, colorWhite, colorWhite35),
                null,
                Shader.TileMode.CLAMP
            )
        }
    }

    override fun drawOverpastWallAndStartLine(canvas: Canvas) {
        drawOverpastWall(canvas)

        // Same as drawStartLine, but with extra gradient color on it
        mStartLinePaint.shader = mStartLineLinearGradient
        mStartLinePaint.color = mLocalPitchIndicatorColor
        mStartLinePaint.isAntiAlias = true
        mStartLinePaint.strokeWidth = 3f
        canvas.drawLine(mCenterXOfStartPoint, 0f, mCenterXOfStartPoint, height.toFloat(), mStartLinePaint)
    }
}